src/psyclone/domain/gocean/gocean_constants.py

```diff
@@ -55,23 +54,29 @@ def __init__(self):
         if GOceanConstants.HAS_BEEN_INITIALISED:
             return
 
+        api_config = Config.get().api_conf("gocean1.0")
```

Comment:
This line causes the first problem:
while loading the config file (after processing command line options in generate.py), the gocean parts tries to `add_bounds` in `GOLoop`. GOLoop needs the gocean_constants.  But now the gocean_constants are trying to get the api_config which is currently being loaded,
I don't see api_config being used here, and (at least the first failing test) passes if I remove this line.

Reply:
Thanks @hiker. I've removed this and turned the reading of access type metadata into a static method as suggested in teams.